streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -1524,6 +1527,12 @@ public void cleanUp() {
      *                                    an InvalidStateStoreException is thrown upon store access.
      */
     public <T> T store(final StoreQueryParameters<T> storeQueryParameters) {
+        synchronized (stateLock) {
+            if (state == State.CREATED) {
+                throw new StreamsNotStartedException("KafkaStreams is not started, you can retry and wait until to running.");
```

Comment:
Technically we don't need to wait for it to be running, they just need to start it:
```suggestion
                throw new StreamsNotStartedException("KafkaStreams has not been started, you can retry after calling start()");
```